From ea0257e1ab7adf5489353c3306b3b00474645248 Mon Sep 17 00:00:00 2001
From: Brock Noland <brock@apache.org>
Date: Fri, 27 Feb 2015 22:49:29 +0000
Subject: [PATCH 0068/1164] HIVE-9803 - SparkClientImpl should not attempt
 impersonation in CLI mode [Spark Branch] (Brock
 via Chao)

git-svn-id: https://svn.apache.org/repos/asf/hive/branches/spark@1662841 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 9f1cceaa7068c893ab2c0980268c64a75f3bf298)
---
 .../apache/hive/spark/client/SparkClientImpl.java  |   11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
index 44549c7..5de7c2e 100644
--- a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
+++ b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
@@ -367,9 +367,14 @@ public void run() {
       }
 
       if (hiveConf.getBoolVar(HiveConf.ConfVars.HIVE_SERVER2_ENABLE_DOAS)) {
-        argv.add("--proxy-user");
         try {
-          argv.add(Utils.getUGI().getShortUserName());
+          String currentUser = Utils.getUGI().getShortUserName();
+          // do not do impersonation in CLI mode
+          if (!currentUser.equals(System.getProperty("user.name"))) {
+            LOG.info("Attempting impersonation of " + currentUser);
+            argv.add("--proxy-user");
+            argv.add(currentUser);
+          }
         } catch (Exception e) {
           String msg = "Cannot obtain username: " + e;
           throw new IllegalStateException(msg, e);
@@ -400,7 +405,7 @@ public void run() {
         argv.add(String.format("%s=%s", hiveSparkConfKey, value));
       }
 
-      LOG.debug("Running client driver with argv: {}", Joiner.on(" ").join(argv));
+      LOG.info("Running client driver with argv: {}", Joiner.on(" ").join(argv));
 
       ProcessBuilder pb = new ProcessBuilder(argv.toArray(new String[argv.size()]));
       if (isTesting != null) {
-- 
1.7.9.5

